lib/services/local/desktops.go

```diff
@@ -36,6 +36,28 @@ func NewWindowsDesktopService(backend backend.Backend) *WindowsDesktopService {
 	return &WindowsDesktopService{Backend: backend}
 }
 
+func (s *WindowsDesktopService) GetNonADDesktopsCount(ctx context.Context) (uint64, error) {
+	startKey := backend.Key(windowsDesktopsPrefix, "")
+	result, err := s.GetRange(ctx, startKey, backend.RangeEnd(startKey), backend.NoLimit)
+	if err != nil {
+		return 0, trace.Wrap(err)
+	}
+
+	count := uint64(0)
+	for _, item := range result.Items {
+		desktop, err := services.UnmarshalWindowsDesktop(item.Value,
+			services.WithResourceID(item.ID), services.WithExpires(item.Expires))
+		if err != nil {
+			return 0, trace.Wrap(err)
+		}
+		if desktop.NonAD() {
+			count++
+		}
+	}
```

Comment:
A no-limit GetRange is rarely what you should do, unless you know that there's only a handful of values in the range.

```suggestion
	stream := backend.StreamRange(ctx, s, startKey, backend.RangeEnd(startKey), defaults.DefaultChunkSize)
	var count uint64
	for stream.Next() {
		item := stream.Item()
		desktop, err := services.UnmarshalWindowsDesktop(item.Value,
			services.WithResourceID(item.ID), services.WithExpires(item.Expires))
		if err != nil {
			stream.Done()
			return 0, trace.Wrap(err)
		}
		if desktop.NonAD() {
			count++
		}
	}
	if err := stream.Done(); err != nil {
		return 0, trace.Wrap(err)
	}
```

Reply:
It might be worth it to also take an optional `maximum useful value` parameter, so we can stop the iteration as soon as we see three non-AD desktops.

Reply:
I've changed it to CheckLimit method, which will do early return if limit is exceeded.

Reply:
Also, I use chunked approach now.